helm-deploy-test/tasks/cf-deploy.sh

```diff
@@ -1,49 +1,80 @@
 #!/bin/bash
 
-set -ex
+set -o errexit -o nounset
 
-#export kube-host details from pool
-set -a; source pool.kube-hosts/metadata; set +a
+# Export kube-host details from pool
+set -o allexport
+source pool.kube-hosts/metadata
+CF_NAMESPACE=cf
+UAA_NAMESPACE=uaa
+UAA_ADMIN_CLIENT_SECRET="$(head -c32 /dev/urandom | base64)"
+set +o allexport
 
-#kube-ready-state-check script
-curl -O https://raw.githubusercontent.com/SUSE/scf/develop/bin/dev/kube-ready-state-check.sh
+# Connect to Kubernetes
+bash -x ci/helm-deploy-test/tasks/common/connect-kube-host.sh
 
-#check kube host readiness to deploy CF
-sshpass -e ssh -o StrictHostKeyChecking=no "${K8S_USER}@${K8S_HOST_IP}" -- \
-    bash -s < kube-ready-state-check.sh
+unzip s3.scf-config/scf-linux-*.zip -d s3.scf-config/
 
-#target the kube cluster
-kubectl config set-cluster --server=http://${K8S_HOST_IP}:${K8S_HOST_PORT} ${K8S_HOSTNAME}
-kubectl config set-context ${K8S_HOSTNAME} --cluster=${K8S_HOSTNAME}
-kubectl config use-context ${K8S_HOSTNAME}
+# Check that the cluster is reasonable
+if test -n "${K8S_USER:-}" -a -n "${SSHPASS:-}" ; then
+    sshpass -e ssh -o StrictHostKeyChecking=no "${K8S_USER}@${K8S_HOST_IP}" -- \
+        bash -s < s3.scf-config/kube-ready-state-check.sh
+else
+    printf "%bWarning: SSH password not supplied, not checking cluster sanity%b\n" \
+        "\033[0;33;1m" "\033[0m" >&2
+fi
 
-unzip s3.scf-config.linux/scf-linux-amd64-* -d scf-config
-
-#Certs generation
-mkdir certs
-pushd scf-config
-./cert-generator.sh -d ${DOMAIN} -n cf -o ../certs
+# Generate certificates
+mkdir certs/
+pushd s3.scf-config/
+./cert-generator.sh -d "${DOMAIN}" -n "${CF_NAMESPACE}" -o ../certs/
 popd
 
-#Deploy UAA
-kubectl create namespace uaa
-helm install scf-config/helm/uaa \
-     --set kube.storage_class.persistent=${STORAGECLASS} \
-     --namespace "uaa" \
-     --values certs/uaa-cert-values.yaml \
-     --set "env.DOMAIN=${DOMAIN}" \
-     --set "env.UAA_ADMIN_CLIENT_SECRET=${UAA_ADMIN_CLIENT_SECRET}" \
-     --set "kube.external_ip=${K8S_HOST_IP}"
-
-#Deploy CF
-kubectl create namespace cf
-helm install scf-config/helm/cf \
-     --set kube.storage_class.persistent=${STORAGECLASS} \
```

Comment:
https://github.com/SUSE/scf/wiki/How-to-Install-SCF#installation-and-deployment-via-helm-including-the-generation-of-certs
```helm install helm/cf \
  --set kube.storage_class.persistent=${STORAGECLASS} \```
?